### PR TITLE
fix: prototype pollution in several npm packages

### PR DESCRIPTION
**Pull request by Kirill89**
The PR is fixing a Prototype Pollution vulnerability in:

- https://www.npmjs.com/package/lodash.defaultsdeep
- https://www.npmjs.com/package/lodash.merge
- https://www.npmjs.com/package/lodash.mergewith

You can see details about similar vulnerability here: https://snyk.io/vuln/SNYK-JS-LODASH-73638

### CODE REVIEW

File: lodash.merge/index.js [hunk 1]
```diff
@@ -1001,7 +1001,7 @@ function baseMergeDeep(object, source, key, srcIndex, mergeFunc, customizer, sta
       if (isArguments(objValue)) {
         newValue = toPlainObject(objValue);
       }
-      else if (!isObject(objValue) || (srcIndex && isFunction(objValue))) {
+      else if (!isObject(objValue) || isFunction(objValue)) {
```
Comment (Kirill89) [inline lodash.merge/index.js]: This is backport from new version https://github.com/snyk/lodash/blob/4.17.12-pre/lodash.js#L3668

File: lodash.mergewith/index.js [hunk 2]
```diff
@@ -1001,7 +1001,7 @@ function baseMergeDeep(object, source, key, srcIndex, mergeFunc, customizer, sta
       if (isArguments(objValue)) {
         newValue = toPlainObject(objValue);
       }
-      else if (!isObject(objValue) || (srcIndex && isFunction(objValue))) {
+      else if (!isObject(objValue) || isFunction(objValue)) {
```
Comment (Kirill89) [inline lodash.mergewith/index.js]: This is backport from new version https://github.com/snyk/lodash/blob/4.17.12-pre/lodash.js#L3668

File: lodash.defaultsdeep/index.js [hunk 3]
```diff
@@ -1130,6 +1130,26 @@ function baseMerge(object, source, srcIndex, customizer, stack) {
...
+/**
+ * Gets the value at `key`, unless `key` is "__proto__" or "constructor".
+ *
+ * @private
+ * @param {Object} object The object to query.
+ * @param {string} key The key of the property to get.
+ * @returns {*} Returns the property value.
+ */
+function safeGet(object, key) {
+  if (key === 'constructor' && typeof object[key] === 'function') {
```
Comment (Kirill89) [inline lodash.defaultsdeep/index.js]: This is fix from the other PR: https://github.com/lodash/lodash/pull/4336

File: lodash.defaultsdeep/index.js [hunk 4]
```diff
@@ -1146,8 +1166,8 @@ function baseMerge(object, source, srcIndex, customizer, stack) {
  *  counterparts.
  */
 function baseMergeDeep(object, source, key, srcIndex, mergeFunc, customizer, stack) {
-  var objValue = object[key],
-      srcValue = source[key],
+  var objValue = safeGet(object, key),
```
Comment (Kirill89) [inline lodash.defaultsdeep/index.js]: This is backport from new version: https://github.com/snyk/lodash/blob/4.17.12-pre/lodash.js#L3624

**Comment by jdalton** [comment]
Thank you @Kirill89!

**Comment by schmod** [comment]
Is there a path forward for getting these published?

**Pull request merged**